jira/client.py

```diff
@@ -2963,6 +2963,32 @@ def resolution(self, id: str) -> Resolution:
 
     # Search
 
+    @overload
+    def search_issues(
+        self,
+        jql_str: str,
+        startAt: int,
+        maxResults: int,
+        validate_query: bool,
+        fields: Optional[Union[str, List[str]]],
+        expand: Optional[str],
+        json_result: Literal[False],
+    ) -> ResultList[Issue]:
+        ...
+
+    @overload
+    def search_issues(
+        self,
+        jql_str: str,
+        startAt: int,
+        maxResults: int,
+        validate_query: bool,
```

Comment:
Let me know if you figure out how to do this so mypy is happy.

I believe it was this issue I was running into when I first started adding typehints for these chunkier functions
https://github.com/python/mypy/issues/6580